packages/utils/src/ast.ts

```diff
@@ -0,0 +1,300 @@
+      if (node.type === 'FunctionExpression' && node.extra?.parenthesized) {
+        node.params = node.params.filter(
+          (param) => !(param.type === 'Identifier' && param.name === 'Buffer'),
+        );
+      }
+    },
+
+    CallExpression(path) {
+      const { node } = path;
+
+      // Replace `eval(foo)` with `(1, eval)(foo)`.
+      if (node.callee.type === 'Identifier' && node.callee.name === 'eval') {
+        path.replaceWith(
+          evalWrapper({
+            REF: node.callee,
+            ARGS: node.arguments,
+          }) as Node,
+        );
+      }
+    },
+
+    MemberExpression(path) {
+      const { node } = path;
+
+      // Replace `object.eval(foo)` with `(1, object.eval)(foo)`.
+      if (
+        node.property.type === 'Identifier' &&
+        node.property.name === 'eval' &&
+        // If the expression is already wrapped we can ignore it
+        path.parent.type !== 'SequenceExpression'
+      ) {
+        path.replaceWith(
+          objectEvalWrapper({
+            OBJECT: node.object,
+            REF: node.property,
+          }) as any,
+        );
+      }
+    },
+
+    Identifier(path) {
+      const { node } = path;
+
+      // Insert `regeneratorRuntime` global if it's used in the code.
+      if (node.name === 'regeneratorRuntime') {
+        const program = path.findParent(
+          (parent) => parent.node.type === 'Program',
+        );
+
+        // We know that `program` is a Program node here, but this keeps
+        // TypeScript happy.
+        if (program?.node.type === 'Program') {
+          const body = program.node.body[0];
+
+          // This stops it from inserting `regeneratorRuntime` multiple times.
+          if (
+            body.type === 'VariableDeclaration' &&
+            (body.declarations[0].id as Identifier).name ===
+              'regeneratorRuntime'
+          ) {
+            return;
+          }
+
+          program?.node.body.unshift(regeneratorRuntimeWrapper());
+        }
+      }
+    },
+
+    TemplateLiteral(path) {
+      const { node } = path;
+
+      // This checks if the template literal was visited before. Without this,
+      // it would cause an infinite loop resulting in a stack overflow. We can't
+      // skip the path here, because we need to visit the children of the node.
+      if (path.getData('visited')) {
+        return;
+      }
+
+      // Break up tokens that could be parsed as HTML comment terminators, or
+      // `import()` statements.
+      // For reference:
+      // - https://github.com/endojs/endo/blob/70cc86eb400655e922413b99c38818d7b2e79da0/packages/ses/error-codes/SES_HTML_COMMENT_REJECTED.md
+      // - https://github.com/MetaMask/snaps-skunkworks/issues/505
+      const expressions = node.quasis.reduce<Expression[]>(
+        (acc, quasi, index) => {
+          // Note: Template literals have two variants, "cooked" and "raw". Here
+          // we just use the raw version, but this might break code that uses
```

Comment:
Do we have an idea of how to solve it?

Reply:
No, the documentation on `raw` versus `cooked` isn't great.